xilem/examples/calc.rs

```diff
@@ -0,0 +1,323 @@
+            data.operation
+                .map(|operation| display_label(operation.as_str())),
+            display_label(data.numbers[1].as_ref()),
+            data.result.is_some().then(|| display_label("=")),
+            data.result
+                .as_ref()
+                .map(|result| display_label(result.as_ref())),
+            FlexSpacer::Flex(0.1),
+        ))
+        .flex(1.0),
+        FlexSpacer::Fixed(10.0),
+        // Top row
+        flex_row((
+            expanded_button("CE", |data: &mut Calculator| {
+                data.clear_entry();
+            })
+            .flex(1.),
+            expanded_button("C", |data: &mut Calculator| data.clear_all()).flex(1.),
+            expanded_button("DEL", |data: &mut Calculator| data.on_delete()).flex(1.),
+            operator_button(MathOperator::Divide).flex(1.),
+        ))
+        .flex(1.0),
+        // 7 8 9 X
+        flex_row((
+            digit_button("7").flex(1.),
+            digit_button("8").flex(1.),
+            digit_button("9").flex(1.),
+            operator_button(MathOperator::Multiply).flex(1.),
+        ))
+        .flex(1.0),
+        // 4 5 6 -
+        flex_row((
+            digit_button("4").flex(1.),
+            digit_button("5").flex(1.),
+            digit_button("6").flex(1.),
+            operator_button(MathOperator::Subtract).flex(1.),
+        ))
+        .flex(1.0),
+        // 1 2 3 +
+        flex_row((
+            digit_button("1").flex(1.),
+            digit_button("2").flex(1.),
+            digit_button("3").flex(1.),
+            operator_button(MathOperator::Add).flex(1.),
+        ))
+        .flex(1.0),
+        // bottom row
+        flex_row((
+            expanded_button("±", |data: &mut Calculator| data.negate()).flex(1.),
+            digit_button("0").flex(1.),
+            digit_button(".").flex(1.),
+            expanded_button("=", |data: &mut Calculator| data.on_equals()).flex(1.),
+        ))
+        .flex(1.0),
+    ))
+    .gap(GRID_GAP)
+    .cross_axis_alignment(CrossAxisAlignment::Fill)
+    .main_axis_alignment(MainAxisAlignment::End)
+    .must_fill_major_axis(true)
+}
+
+/// Creates a horizontal centered flex row designed for the display portion of the calculator.
+pub fn centered_flex_row<Seq, Marker>(sequence: Seq) -> Flex<Seq, Marker> {
+    flex(sequence)
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Center)
+        .main_axis_alignment(MainAxisAlignment::Start)
+        .gap(5.)
+}
+
+/// Creates a horizontal filled flex row designed to be used in a grid.
+pub fn flex_row<Seq, Marker>(sequence: Seq) -> Flex<Seq, Marker> {
+    flex(sequence)
+        .direction(Axis::Horizontal)
+        .cross_axis_alignment(CrossAxisAlignment::Fill)
+        .main_axis_alignment(MainAxisAlignment::SpaceEvenly)
+        .gap(GRID_GAP)
+}
+
+/// Returns a label intended to be used in the calculator's top display.
+/// It sets the text size to DISPLAY_FONT_SIZE.
```

Comment:
```suggestion
```
Similarly here - comments which trivially restate the code aren't that useful in examples, unless you're commenting on the exact line in question.

We could also have something like:
```suggestion
/// The default text size is out of proportion for this use case.
```
I.e. explain why we use a bigger text size.